state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -134,31 +134,29 @@ impl AptosNetDataClient {
         *self.global_summary_cache.write() = aggregate;
     }
 
-    /// Choose a connected peer that can service the given request. Returns an
-    /// error if no such peer can be found.
+    /// Choose a connected peer that can service the given request.
+    /// Returns an error if no such peer can be found.
     fn choose_peer_for_request(
         &self,
         request: &StorageServiceRequest,
     ) -> Result<PeerNetworkId, Error> {
-        // Try to send the request to a priority peer that can service it
-        let internal_peer_states = self.peer_states.read();
-        let priority_connected_peers = self.get_priority_connected_peers()?;
-        let mut serviceable_peers = priority_connected_peers
-            .into_iter()
-            .filter(|peer| internal_peer_states.can_service_request(peer, request))
-            .collect::<Vec<_>>();
-
-        // If there are no priority serviceable peers, try to send the request
-        // to a regular peer that can service it.
-        if serviceable_peers.is_empty() {
-            let regular_connected_peers = self.get_regular_connected_peers()?;
-            serviceable_peers = regular_connected_peers
-                .into_iter()
-                .filter(|peer| internal_peer_states.can_service_request(peer, request))
-                .collect::<Vec<_>>();
-        }
+        // Identify the peers that can service the request
+        let serviceable_peers = if request.is_data_subscription_request() {
```

Comment:
How about the storage summary request? Do you also want to prioritize sending that to the prioritized peer?

Reply:
Sure! Note: there's no clients who use this functionality, so it feels a little obsolete. But, it's nice from a consistency point of view 😄 

Also: I've added another commit to this PR. Nothing big, but it moves calculation of the overall aggregate summary out of the poller and into the main loop. This makes more sense semantically, i.e., each poller should only update the data for the peer. The main loop should be responsible for calculating the overall summary. I think this is cleaner :)

Reply:
> Note: there's no clients who use this functionality, so it feels a little obsolete. But, it's nice from a consistency point of view

Can you clarify this? I was under impression that in SSv2, the client (or the poller) sends periodic storage summary request to the server?

Reply:
> Can you clarify this?

Yeah, so the poller selects the peers it wishes to poll and then polls those peers directly using `send_request_to_peer_and_decode`. This method (`choose_peer_for_request`) is only used by clients (e.g., the data streaming service) who wish to get data and don't care where it comes from. We don't currently have any users of `choose_peer_for_request` where the request is a storage summary request. Does that make sense?